modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/randomsql/RandomQuerySupplier.java

```diff
@@ -80,151 +92,211 @@ public RandomQuerySupplier(Schema schema, int seed) {
+    private Select rndSingleTableSelect(RandomisedQueryContext rndQryCtx) {
+        pickRndTables(rndQryCtx, 1);
 
-        return new Select().from(from).where(where);
+        assert rndQryCtx.scopeTables().size() == 1;
+
+        return new Select(rndColList(rndQryCtx))
+            .from(new TableList(rndQryCtx.scopeTables()))
+            .where(rndQueryFilter(rndQryCtx));
     }
 
     /**
-     * Generates random FROM expression.
+     * Generates random query filter.
      *
      * @param rndQryCtx Context of randomised query.
-     * @return Ast representing FROM expression.
+     * @return Ast representing query filter.
      */
-    private Ast rndFrom(RandomisedQueryContext rndQryCtx) {
-        List<TableRef> tbls = rndTables(rndQryCtx.schema());
+    private Ast rndQueryFilter(RandomisedQueryContext rndQryCtx) {
+        List<TableRef> scopeTbls = rndQryCtx.scopeTables();
+
+        Ast cond1 = rndTableFilter(rndQryCtx, scopeTbls.get(rnd.nextInt(scopeTbls.size())));
 
-        assert !F.isEmpty(tbls);
+        if (dice.roll() <= 2)
+            return cond1;
 
-        if (tbls.size() == 1 || rndWithRatio(5)) {
-            tbls.forEach(rndQryCtx::addScopeTable);
+        Ast cond2 = rndTableFilter(rndQryCtx, scopeTbls.get(rnd.nextInt(scopeTbls.size())));
 
-            return new TableList(tbls);
+        return new BiCondition(cond1, cond2, dice.roll() <= 2 ? Operator.AND : Operator.OR);
+    }
+
+    /**
+     * Generates random filter for specified table.
+     *
+     * @param rndQryCtx Context of randomised query.
+     * @param tbl Table that requires filter.
+     * @return Ast representing table filter.
+     */
+    private Ast rndTableFilter(RandomisedQueryContext rndQryCtx, TableRef tbl) {
+        Ast rigthOp;
+        if (dice.roll() == 1)
+            rigthOp = new Const(Integer.toString(rnd.nextInt(100)));
+        else {
+            rigthOp = new Const("?");
+
+            rndQryCtx.addQueryParam(rnd.nextInt(100));
         }
 
-        TableRef leftTbl = tbls.get(0);
+        return new BiCondition(
+            rndNumericColumn(tbl),
+            rigthOp,
+            Operator.EQUALS
+        );
+    }
 
-        Ast from = leftTbl;
+    /**
+     * Picks N random table from the schema and adds them to the scope.
+     *
+     * @param rndQryCtx Random query context.
+     * @param tblCnt Tables count.
+     */
+    private void pickRndTables(RandomisedQueryContext rndQryCtx, int tblCnt) {
+        pickRndTables(rndQryCtx, tblCnt, false);
+    }
 
-        rndQryCtx.addScopeTable(leftTbl);
+    /**
+     * Picks N random table from the schema and adds them to the scope.
+     *
+     * @param rndQryCtx Random query context.
+     * @param tblCnt Tables count.
+     * @param allowDublicates Whether to allow dublicates or not.
+     */
+    private void pickRndTables(RandomisedQueryContext rndQryCtx, int tblCnt, boolean allowDublicates) {
+        List<Table> tbls = new ArrayList<>(schema.tables());
```

Comment:
Why we use local schema instead of tndQryCtx.schema? And why we need same schema in RandomisedQueryContext if we never use it?
It hard to track which schema is preferable in different situations and I hope we can remove one or comment the difference.

Reply:
schema should be part of RandomQuerySupplier only. So I've remove obsolete field inside RandomisedQueryContext